### PR TITLE
Don't create TensorBoard logs during prediction

### LINKED ISSUE #5617: Tensorboard log files are created during prediction as well
**Rasa version**: 1.9.5

**Python version**: 3.6.5

**Operating system** (windows, osx, ...): MacOS

**Issue**: Since the call to `self._set_up_tensorboard_writer(tensorboard_log_level, tensorboard_log_dir)` is made in the constructor of `RasaModel` class, these files are created during prediction time as well, for example when `rasa test` is run. 

**Command or request that led to error**: `rasa test nlu`

### PR DESCRIPTION
**Pull request by tabergma**
**Proposed changes**:
Don't create TensorBoard log files during prediction.

closes #5617

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/utils/tensorflow/models.py [hunk 1]
```diff
@@ -103,6 +103,10 @@ def fit(
     ) -> None:
         """Fit model data"""
 
+        if not silent:
```
Comment (dakshvar22) [inline rasa/utils/tensorflow/models.py:106]: Why do we need this `if` construct? Isn't `fit()` called only during training?
Also, if its necessary then we should rename silent to something more meaningful?

Comment (dakshvar22) [inline rasa/utils/tensorflow/models.py:106]: Why do we need this if construct? Isn't fit() called only during training?
Also, if its necessary then we should rename silent to something more meaningful?

Comment (tabergma) [reply rasa/utils/tensorflow/models.py:106]: It is also called during load, that's why I added the comment `# don't setup tensorboard writers when training during loading`. We can maybe rename it or add another parameter.

Comment (dakshvar22) [reply rasa/utils/tensorflow/models.py:106]: Ahh, right its called in load to pass through a dummy batch.
Yes, maybe adding a separate parameter for it is better.

**Pull request merged**